manifests/server/role.pp

```diff
@@ -166,7 +166,7 @@
         $pwd_hash_sql = postgresql::postgresql_password(
           $username,
           $password_hash,
```

Comment:
Calling `postgresql::postgresql_password()` with a variable `$password_hash` as the `password` parameter look wrong: as far as I can see, `postgresql::server::role` only has a `password_hash` parameter, so I would not expect it to hash anything itself (and be passed an already hashed password).

The `postgresql::server::db` class has a `password` parameter which is copied to the `password_hash` parameter of `postgresql::server::role` verbatim.  This is inconsistent, but I think this is for backward compatibility since when I check my (old) code I have things like:

```puppet
postgresql::server::db { $taiga::back::db_name:
  user     => $taiga::back::db_user,
  password => postgresql_password($taiga::back::db_user, $taiga::back::db_password),
}
```

The test suite mostly checks that values are sensitive ones, so basically that they match `Sensitive [value redacted]`, but not the actual unwrapped value.  [I opened a PR](https://github.com/puppetlabs/rspec-puppet/pull/42) to unwrap Sensitive values when running the test suite and as you guessed on Slack, some content seems to be wrapped twice.

Reply:
Yes, the hashing of a hash was also my first guess but it works since it returns the same value as long as the hashtype and salt does match.
Already have an idea to do a check like that to prevent duplicate hashing:
```
unless $password_hash_unsensitive =~ /^md5[0-9a-f]{32}$/ and $password_hash_unsensitive =~ /^scram-sha-256:/ { 
  $password_sql = postgresql_password([...])
} else {
  $password_sql = $password_hash_unsensitive
}
```
But i would say i create an additional PR since this PR is bug fixing and the duplicate hashing is more or less a maintenance change.

Reply:
Or just edit lines 182 and 183??

eg.
```puppet
$pw_command = "ALTER ROLE \"${username}\" ENCRYPTED PASSWORD '${pwd_hash_sql}'"
```

becomes
```puppet
$pw_command = "ALTER ROLE \"${username}\" ENCRYPTED PASSWORD '${pwd_hash_sql.unwrap}'"
```

Reply:
Sure but then we should also update the `Deferred` way to use Sensitive and rework to remove this, or?
https://github.com/puppetlabs/puppetlabs-postgresql/blob/ad330fc69fbaa54104934d7e0e05cf03655dd2ec/manifests/server/role.pp#L45-L49

Reply:
That part predates the Puppet 6.24 minimum version bound and can indeed be simplified.